File: test/semgrep-test/vulnerable.py
# 测试Semgrep规则检测

import sqlite3
import os
import subprocess
import pickle
import hashlib

# 应该触发 python-sql-injection 规则
def unsafe_query(user_input):
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    query = f"SELECT * FROM users WHERE id = {user_input}"
    cursor.execute(query)
    return cursor.fetchall()

# 应该触发 python-command-injection 规则
def run_command(user_input):
    os.system(user_input)

# 应该触发 python-hardcoded-secret 规则
PASSWORD = "my-secret-password-123"

# 应该触发 python-deserialization 规则
def load_data(user_input):
    return pickle.loads(user_input)

# 应该触发 python-weak-crypto 规则
def hash_password(password):
    return hashlib.md5(password.encode()).hexdigest()

File: test/semgrep-test/vulnerable.js
// 测试Semgrep规则检测

// 应该触发 javascript-sql-injection 规则
function unsafeQuery(userInput) {
    const query = "SELECT * FROM users WHERE id = '" + userInput + "'";
    return db.query(query);
}

// 应该触发 javascript-xss-reflected 规则
function renderUserContent(userInput) {
    document.getElementById('output').innerHTML = userInput;
}

// 应该触发 javascript-hardcoded-secret 规则
const API_KEY = "sk-1234567890abcdef";

// 应该触发 javascript-eval-usage 规则
eval(userCode);

// 应该触发 javascript-prototype-pollution 规则
obj["__proto__"][key] = value;